n3fit/src/n3fit/backends/keras_backend/operations.py

```diff
@@ -107,36 +106,31 @@ def numpy_to_tensor(ival, **kwargs):
 
 # f(x: tensor) -> y: tensor
 def batchit(x, batch_dimension=0, **kwarg):
-    """ Add a batch dimension to tensor x """
+    """Add a batch dimension to tensor x"""
     return tf.expand_dims(x, batch_dimension, **kwarg)
 
 
 # layer generation
-def numpy_to_input(numpy_array, no_reshape=False, name=None):
+def numpy_to_input(
+    numpy_array: np.ndarray, name: str = None,
+):
     """
-    Takes a numpy array and generates a Input layer.
-    By default it adds a batch dimension (of size 1) so that the shape of the layer
-    is that of the array
+    Takes a numpy array and generates an Input layer with the same shape,
+    but with a batch dimension (of size 1) added.
 
     Parameters
     ----------
         numpy_array: np.ndarray
-        no_reshape: bool
-            if true, don't add batch dimension, take the first dimension of the array as the batch
-        name: bool
+        name: str
             name to give to the layer
     """
-    if no_reshape:
-        batched_array = numpy_array
-        batch_size = numpy_array.shape[0]
-        shape = numpy_array.shape[1:]
-    else:
-        batched_array = np.expand_dims(numpy_array, 0)
-        batch_size = 1
-        shape = numpy_array.shape
-    input_layer = Input(batch_size=batch_size, shape=shape, name=name)
+    batched_array = np.expand_dims(numpy_array, 0)
+    shape = list(numpy_array.shape)
+    # set the number of gridpoints to None, otherwise shapes don't show in model.summary
+    shape[0] = None
```

Comment:
I think it's because the regular grid has shape None in this axis while the integration grid has shape 1_000, and the x=1 grid shape 1. It doesn't merge them together as None

Reply:
Ok I'll run those checks.

I never tested the one flavour per network, actually I was wondering if that was still being used, if not I wanted to merge the `generate_dense_per_flavour_network` and `generate_nn` functions. If you're already wondering if it still works it might make sense to remove?

Reply:
No, if it doesn't work it should actually be fixed! While it is not something we actively use, it used to be the way in which the PDF fit was done so it's good to have it if someone would like to reproduce the old ways.
(if you didn't try it don't worry though, I was just wondering)

Reply:
But if it hasn't been used in a long time doesn't it make sense to refer people to an older release? In this case it doesn't matter much, but I can imagine the need to always be backward compatible with everything will force you to make suboptimal choices for new additions, and overcomplicate the code. Not an issue here but just wondering what the policy is.

Reply:
No, I mean if people want to use the newer features (data, theory, whatever) with the older strategy.

And, in any case, we want the architecture to be quite flexible since it is the type of thing that make sense to change for a side project (for example, there's currently people trying with functional forms, in the past we have fitted using a quantum computer). I guess is the kind of thing that might look inelegant but it has been useful in the past so I'd like to keep it. Btw, I just tried it out and seems to work fine at least with the basic_runcard, so the refactor didn't break that :)

Policy-wise the idea is to keep everything compatible one version below, so, to first approximation, nothing that was used for the 4.0 paper should disappear until 5.0 (this is also one of the reasons we are keeping all the C++ around for the time being...)